[expo-ui] Add docs (#34808)
expo/expo: docs/pages/versions/unversioned/sdk/ui.mdx

---
title: UI
description: A set of platform components to build native-feeling UIs.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-ui'
packageName: '@expo/ui'
platforms: ['android', 'ios']
---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { Tabs, Tab } from '~/ui/components/Tabs';

> `@expo/ui` is a set of native input components that allows you to build native-feeling forms.

Reviewer: In the page's description we say native feeling "UIs" and here we are referring to "forms". I think it might lead to confusion so let's update here or the description to match the verbiage.

> > **warning** This library is currently in alpha and subject to breaking changes. It is not available in the Expo Go app.

Reviewer: ```suggestion
> **warning** This library is currently in alpha and subject to breaking changes. It is unavailable in the Expo Go app, and you should use [development builds](/develop/development-builds/introduction/).
```

Let's also add the link and context about development builds.

## Installation

<APIInstallSection />

## Examples

### Button

<Tabs>

> <Tab label="iOS Screenshot">

Reviewer: ```suggestion
<Tab label="iOS">
```

Let's also remove screenshot from every tab label. It's self-explanatory that it's a screenshot and we need only need to mention the platform name.

<ContentSpotlight
  alt="Button component on iOS."
  src="/static/images/expo-ui/button/ios.png"
  className="max-w-[480px]"
/>

</Tab>

> <Tab label="Android Screenshot">

Reviewer: Can we make Android tab the initial tab for all sections where there is an Android example? Relates to the platform order from our writing style guide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#platform-order-when-referencing-android-ios-and-web-or-multiple-platforms

---
title: Expo UI
sidebar_title: UI
description: A set of platform components to build native-feeling UIs.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-ui'
packageName: '@expo/ui'
platforms: ['android', 'ios']
isNew: true
---

import APISection from '~/components/plugins/APISection';
import { APIInstallSection } from '~/components/plugins/InstallSection';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { Tabs, Tab } from '~/ui/components/Tabs';

`@expo/ui` is a set of native input components that allows you to build fully native interfaces with SwiftUI and Jetpack Compose.

> **warning** This library is currently in alpha and subject to breaking changes. It is unavailable in the Expo Go app, and you should use [development builds](/develop/development-builds/introduction/).

## Installation

<APIInstallSection />

## Examples

### Button

<Tabs>

> <Tab label="Android">
> 
> <Tab label="iOS">
> 
> <ContentSpotlight
>   alt="Button component on iOS."
>   src="/static/images/expo-ui/button/ios.png"
>   className="max-w-[480px]"
> />
> 
> </Tab>
> 
> <ContentSpotlight
>   alt="Button component on Android."
>   src="/static/images/expo-ui/button/android.png"
>   className="max-w-[480px]"
> />
> 
> </Tab>

Reviewer 2: ```suggestion
<Tab label="Android">

<ContentSpotlight
  alt="Button component on Android."
  src="/static/images/expo-ui/button/android.png"
  className="max-w-[480px]"
/>

</Tab>

<Tab label="iOS">

<ContentSpotlight
  alt="Button component on iOS."
  src="/static/images/expo-ui/button/ios.png"
  className="max-w-[480px]"
/>

</Tab>

```

<Tab label="Code">

```tsx
import { Button } from '@expo/ui/components/Button';

<Button
  style={{ flex: 1 }}
  onPress={() => {
    setEditingProfile(true);
  }}>
  Edit profile
</Button>;
```

</Tab>

</Tabs>

### Picker (segmented variant)

<Tabs>

> <Tab label="Android">
> 
> <Tab label="iOS">
> 
> <ContentSpotlight
>   alt="Picker component on iOS."
>   src="/static/images/expo-ui/segmentedPicker/ios.png"
>   className="max-w-[480px]"
> />
> 
> </Tab>
> 
> <ContentSpotlight
>   alt="Picker component on Android."
>   src="/static/images/expo-ui/segmentedPicker/android.png"
>   className="max-w-[480px]"
> />
> 
> </Tab>

Reviewer 2: ```suggestion
<Tab label="Android">

<ContentSpotlight
  alt="Picker component on Android."
  src="/static/images/expo-ui/segmentedPicker/android.png"
  className="max-w-[480px]"
/>

</Tab>

<Tab label="iOS">

<ContentSpotlight
  alt="Picker component on iOS."
  src="/static/images/expo-ui/segmentedPicker/ios.png"
  className="max-w-[480px]"
/>

</Tab>

```

<Tab label="Code">

```tsx
import { Picker } from '@expo/ui/components/Picker';

<Picker
  options={['$', '$$', '$$$', '$$$$']}
  selectedIndex={selectedIndex}
  onOptionSelected={({ nativeEvent: { index } }) => {
    setSelectedIndex(index);
  }}
/>;
```

</Tab>

</Tabs>

### Picker (wheel variant) (iOS only)

<Tabs>
<Tab label="iOS">

<ContentSpotlight
  alt="Picker component on iOS."
  src="/static/images/expo-ui/menuPicker/ios.png"
  className="max-w-[480px]"
/>

</Tab>

{/* TODO: Add screenshot on Android once we support the label prop */}

<Tab label="Code">

```tsx
import { Picker } from '@expo/ui/components/Picker';

<Picker
  options={['$', '$$', '$$$', '$$$$']}
  selectedIndex={selectedIndex}
  onOptionSelected={({ nativeEvent: { index } }) => {
    setSelectedIndex(index);
  }}
  variant="wheel"
  style={{
    height: 100,
  }}
/>;
```

</Tab>

</Tabs>

> ### Switch

Reviewer 2: I think we are missing screenshots for the Android `Switch` and checkbox `Switch`

Author: yes, we should support the label prop correctly there first before taking those though

Author: or build a screenshot automation 😅
